src/lib/free-queue/test/free-queue.test.js

```diff
@@ -0,0 +1,162 @@
+import { expect } from 'https://cdnjs.cloudflare.com/ajax/libs/chai/5.1.1/chai.js';
+import { FreeQueue, MAX_CHANNEL_COUNT, RENDER_QUANTUM_FRAMES } from '../free-queue.js';
+
+// Mock WASM module
+const mockWasmModule = {
+  // Simulate memory allocation
+  _malloc: (size) => new ArrayBuffer(size), 
+  // Simulate memory deallocation
+  _free: () => {}, 
+  // Simulate HEAPF32
+  HEAPF32: new Float32Array(1024), 
+};
+
+describe('FreeQueue Class', () => {
+  const bufferLength = 512;
```

Comment:
Let's change this to 1024 to match the FreeQueueSAB tests, so that the performance tests can be compared more directly.